### DISCUSSION TITLE
How to Migrate storage from JanusGraph FoundationDb to JanusGraph Apache Cassandra (cql)

### DISCUSSION THREAD

**Discussion by apeterlic** [JanusGraph/janusgraph#3678] [Q&A] [answered]
Hello,

I am using FoundationDB as backend storage for JanusGraph. Currently, I am migrating to Apache Cassandra (cql). 
How would you recommend I migrate the old data? Is migration supported by JanusGraph or am I required to manually copy the data?

Thanks in advance!

**Comment by hadoopmarc** [comment, marked as answer]
Unfortunately, manually copying the data will be the only way. Assuming a significant amount of data (that is the point of using JanusGraph), parallelization of the copying is a concern and it might be easier to start from the original data outside FoundationDB. Do you have any properties indexed on a mixed indexing backend that could help in leveraging property value ranges for parallelization?

**Reply by apeterlic** [reply]
I appreciate your quick response. Thank you for pointing out the parallelization concern.
I am using mixed indexes and Elasticsearch as an external index backend. Maybe some of them could be helpful. 
Thanks!

**Comment by mad** [comment]
You could replicate stores one by one through ScanJob API. This approach minimize serialization overhead and stay same all internal ids

During replication master graph shouldn't be modified for consistence

Sample implementation might look like this https://gist.github.com/mad/9bf87eadfe782a758b423882e3fdd3cc

**Comment by harshalvaidya10** [comment]
> Hello,
> 
> I am using FoundationDB as backend storage for JanusGraph. Currently, I am migrating to Apache Cassandra (cql). How would you recommend I migrate the old data? Is migration supported by JanusGraph or am I required to manually copy the data?
> 
> Thanks in advance!

Can you tell us how were you able to use FoundationDB as a storage backend with JanusGraph?
